lib/models/users-schema.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+// == EXTERNAL RESOURCES ===============================================
+
+const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+
+// == DEFINE THE USER SCHEMA =============================================
+
+// mongoose schema of users 
+const users = new mongoose.Schema({
+  username: { type: String, required: true, unique: true },
+  password: { type: String, required: true },
+  email: { type: String },
+  role: {
+    type: String,
+    required: true,
+    default: 'user',
+    enum: ['admin', 'editor', 'user']
+  }
+});
+
+/**
+ * pre mongoDB middleware
+ * right before save a record, run this code
+ * bcrypt.hash password
+ */
+ 
+users.pre('save', async function() {
+  if (this.isModified('password')) {
+    this.password = await bcrypt.hash(this.password, 10);
+  }
+});
+ 
+/**
+ * @param creds is the object containing all the credentials for this user
+ * return single record 
```

Comment:
For the JSDocs try declaring the type after the `@param` like `@param {object} creds` the object containing all the credentials(username/password) for this user